frontend/src/components/phones/onboarding/RealPhoneSetup.tsx

```diff
@@ -171,7 +179,7 @@ const RealPhoneVerification = (props: RealPhoneVerificationProps) => {
     useState<boolean>();
 
   const onChange: ChangeEventHandler<HTMLInputElement> = (event) => {
-    setVerificationCode(event.target.value);
+    setVerificationCode(event.currentTarget.value.replace(/[^0-9]/gi, ""));
```

Comment:
Regexes are always somewhat complex to parse, so could you help the reader by adding a comment describing what this does? If I'm following it correctly, my guess would be that it removes everything that's not a number from the string? (Also, I think the `i` flag is not necessary, since numbers are case-insensitive?)

Reply:
That said, I think generally it's considered bad UX practice to modify the user's input while they're inputting it. For example, what sometimes happens for me is that I'm practically blindly typing something, realise I'm mistyping something, and pressing Backspace right away to correct it. With the current setup, that would result in my entering an invalid value.

IIRC, better is to just provide inline validation hints. In this case, I expect that setting `minLength={6}` in addition to the `maxLength` will probably be enough to ensure that the user enters the correct code.

Reply:
<img width="586" alt="image" src="https://user-images.githubusercontent.com/3924990/186557535-07bb34ee-57a0-4656-8abd-c3ccda927bc6.png">
<img width="473" alt="image" src="https://user-images.githubusercontent.com/3924990/186557575-eec7e9df-d0ac-4520-9ebd-cf8ac5783060.png">

I added minLength, pattern, onInvalid and title. Feels a bit better. This gives us a tooltip for when user hovers over the input. This also makes it so that when the user submits this and it doesn't match the pattern, it lets them know "Enter 6-digit code".

Reply:
Ah, I didn't see this remark because I was using the "Changes since last review" view, but again, really clean solution!